src/GraphQL/Internal/AST.hs

```diff
@@ -76,12 +130,12 @@ data OperationDefinition = Query    { getNode :: Node }
                          | Mutation { getNode :: Node }
                            deriving (Eq,Show)
 
-data Node = Node Name [VariableDefinition] [Directive] SelectionSet
+data Node = Node (Maybe Name) [VariableDefinition] [Directive] SelectionSet
```

Comment:
Imagine:

```graphql
{
  me {
    name
  }
}
```

The top-level node has no name. There can only be one such node in a document. I'm open to better ideas on how to represent this.